addOns/postman/src/main/java/org/zaproxy/addon/postman/PostmanParser.java

```diff
@@ -98,16 +102,36 @@ public void importFromUrl(final String url) throws IllegalArgumentException, IOE
         }
 
         String collectionJson = requestor.getResponseBody(uri);
-        importCollection(collectionJson);
+        return importCollection(collectionJson);
     }
 
-    public void importCollection(String collection) throws JsonProcessingException {
+    List<HttpMessage> getHttpMessages(String collection, List<String> errors)
+            throws JsonProcessingException {
         PostmanCollection postmanCollection = parse(collection);
-
         List<HttpMessage> httpMessages = new ArrayList<>();
-        extractHttpMessages(postmanCollection.getItem(), httpMessages);
 
-        requestor.run(httpMessages);
+        extractHttpMessages(postmanCollection.getItem(), httpMessages, errors);
+
+        return httpMessages;
+    }
+
+    public boolean importCollection(String collection) throws JsonProcessingException {
+        List<String> errors = new ArrayList<>();
+        List<HttpMessage> httpMessages = getHttpMessages(collection, errors);
+
+        requestor.run(httpMessages, errors);
+
+        logErrors(errors);
+
+        return errors.size() == 0;
+    }
+
+    private static void logErrors(List<String> errors) {
```

Comment:
- `outputErrors`? Not logging them.
 - This should only be done if there's view.

Reply:
And the API should return them.